src/utils/connection_manager.py

```diff
@@ -0,0 +1,25 @@
+class Borg:
+    _shared_state = {}
+
+    def __init__(self):
+        self.__dict__ = self._shared_state
+
+
+class ConnectionManager(Borg):
+
```

Comment:
Basically this is a pool manager. Pools are not offered in the basic driver of crate and psql. It would be good to explore a proper db pool.